### PR TITLE
Use python 3.5 for doctest

### PR DESCRIPTION
**Pull request by sonots**
Related: https://github.com/chainer/chainer-test/pull/342

Fixed doctest to pass with python 3.5.

### CODE REVIEW

**Comment by sonots** [comment]
```
Doctest summary
===============
   60 tests
    0 failures in tests
    0 failures in setup code
    0 failures in cleanup code
```

**Comment by okuta** [comment]
LGTM!

File: docs/source/reference/difference.rst [hunk 1]
```diff
@@ -90,6 +90,6 @@ If these types were returned, it would be required to synchronize between GPU an
 If you want to use scalar values, cast the returned arrays explicitly.
 
   >>> type(np.sum(np.arange(3)))
-  <type 'numpy.int64'>
```
Comment (okuta) [inline docs/source/reference/difference.rst]: How about using `isinstance` ?
It is accepted by Python2 and Python3.

Comment (sonots) [reply docs/source/reference/difference.rst]: Used `type() ==` no to change much, but works with both 2 and 3.

**Comment by sonots** [comment]
Number of tests increased by unknown reasons ....

```
Doctest summary
===============
   95 tests
    0 failures in tests
    0 failures in setup code
    0 failures in cleanup code
```

**Comment by okuta** [comment]
Thanks!

**Pull request merged**